CRM/Contact/BAO/Contact.php

```diff
@@ -3056,6 +3056,7 @@ public static function contextMenu($contactId = NULL) {
             'ref' => $values['ref'],
             'class' => $values['class'] ?? NULL,
             'key' => $values['key'],
+            'weight' => CRM_Core_Action::getWeight($values['weight']) ?? $values['weight'],
```

Comment:
What's going on here? Should this be 
```suggestion
            'weight' => CRM_Core_Action::getWeight($values['key']) ?? $values['weight'],
```

Reply:
Actually for those ones the weight should be set so I am just using the set value - surrounded in this if
```
  // build directly accessible action menu.
        if (in_array($values['ref'], [
          'view-contact',
          'edit-contact',
        ])) {
```

Reply:
Ok I just don't understand how passing `$values['weight']` into `CRM_Core_Action::getWeight()` could possibly make sense. That function expects a constant like `CRM_Core_Action::VIEW` not a weight like `10`.